atmcfg/atmcfg_test.go

```diff
@@ -693,3 +694,67 @@ func TestReclaimFreeSpace(t *testing.T) {
 		}
 	})
 }
+
+func TestReclaimFreeSpaceForProcessesByClusterName(t *testing.T) {
+	lastCompact := time.Now().Format(time.RFC3339)
+	t.Run("replica set", func(t *testing.T) {
+		config := automationConfigWithOneReplicaSet(clusterName, true)
+		err := ReclaimFreeSpaceForProcessesByClusterName(config, clusterName, lastCompact, []string{"host0:27017"})
+		if err != nil {
+			t.Fatalf("ReclaimFreeSpaceForProcessesByClusterName() returned an unexpected error: %v", err)
+		}
+
+		if config.Processes[0].LastCompact == "" {
+			t.Errorf("Got = %#v", config.Processes[0].LastRestart)
+		}
+	})
+
+	t.Run("sharded cluster - two processes", func(t *testing.T) {
+		config := automationConfigWithOneShardedCluster(clusterName, true)
+		err := ReclaimFreeSpaceForProcessesByClusterName(config, clusterName, lastCompact, []string{"host0:27017", "host2:27018"})
+		if err != nil {
+			t.Fatalf("ReclaimFreeSpaceForProcessesByClusterName() returned an unexpected error: %v", err)
+		}
+		for i := range config.Processes {
+			if isLastCompactEmpty(config.Processes[i], "host2", 27017) {
+				t.Errorf("ReclaimFreeSpace\n got=%#v", config.Processes[i].LastRestart)
+			}
+			if isLastCompactEmpty(config.Processes[i], "host0", 27018) {
+				t.Errorf("ReclaimFreeSpace\n got=%#v", config.Processes[i].LastRestart)
+			}
+		}
+	})
+	t.Run("restart entire sharded cluster", func(t *testing.T) {
+		config := automationConfigWithOneShardedCluster(clusterName, true)
+		err := ReclaimFreeSpaceForProcessesByClusterName(config, clusterName, lastCompact, nil)
+		if err != nil {
+			t.Fatalf("ReclaimFreeSpaceForProcessesByClusterName() returned an unexpected error: %v", err)
+		}
+		for i := range config.Processes {
+			if isLastCompactEmpty(config.Processes[i], "", -1) {
+				t.Errorf("ReclaimFreeSpace\n got=%#v", config.Processes[i].LastRestart)
+			}
+		}
+	})
+	t.Run("provide a process that does not exist", func(t *testing.T) {
+		config := automationConfigWithOneShardedCluster(clusterName, true)
+		err := ReclaimFreeSpaceForProcessesByClusterName(config, clusterName, lastCompact, []string{"hostTest:21021"})
+		if !errors.Is(err, ErrProcessNotFound) {
+			t.Fatalf("Got = %#v, want = %#v", err, ErrProcessNotFound)
+		}
+
+		for i := range config.Processes {
+			if config.Processes[i].LastRestart != "" {
+				t.Errorf("Got = %#v, want = %#v", config.Processes[i].LastRestart, "")
+			}
+		}
+	})
+}
+
+func isLastCompactEmpty(process *opsmngr.Process, hostname string, portN int) bool {
+	if hostname != "" && process.Args26.NET.Port != portN && process.Hostname != hostname {
+		return false
+	}
+
+	return process.ProcessType == "mongod" && process.LastCompact == ""
+}
```

Comment:
[nit] make this a full on helper, pass `t` and you can do `t.Errorf("ReclaimFreeSpace\n got=%#v", config.Processes[i].LastRestart)` inside the method and just call `isLastCompactEmpty` without checking the return value